pkg/log/logger.go

```diff
@@ -194,32 +187,14 @@ func (l *gosoLogger) log(level int, msg string, args []interface{}, loggedErr er
 	}
 }
 
-func (l *gosoLogger) shouldLogToChannels(current string, channels []string) bool {
-	if len(channels) == 0 {
-		return true
-	}
-
-	for _, channel := range channels {
-		if channel == "*" {
-			return true
-		}
-
-		if channel == current {
-			return true
-		}
-
-		if !strings.HasPrefix(channel, "!") {
-			continue
-		}
-
-		channel = strings.TrimPrefix(channel, "!")
-
-		if channel == current {
-			return false
+func (l *gosoLogger) shouldLog(current string, level int, h Handler) bool {
+	for _, channel := range h.Channels() {
```

Comment:
I guess some kind of map to look up the channel name would be useful here, otherwise you have to search all channels every time you log something. I think that would also work well directly in the config like this:
```yml
channels:
    metrics:
        level: info
        enabled: true
    kernel:
        level: warning
```